apps/examples/avs_test/avs_test_main.cpp

```diff
@@ -74,66 +76,207 @@
+	} while (0)
+
+#define WIFI_CONNECTED_WAIT() \
+	do {\
+		std::unique_lock<std::mutex> lock(sg_wifiMutex);\
+		sg_wifiCondv.wait_for(lock, std::chrono::seconds(10), []{return sg_wifiIsConnected;});\
+	} while (0)
+
+static void wifi_sta_connected(wifi_manager_result_e result) {
+	printf("[%s] result %d\n", __FUNCTION__, (int)result);
+	WIFI_CONNECTED_NOTIFY();
 }
 
-static wifi_manager_cb_s wifi_callbacks = {
-	wifi_sta_connected,
-	wifi_sta_disconnected,
-	NULL,
-	NULL,
-	NULL,
-};
+static void wifi_sta_disconnected(wifi_manager_disconnect_e result) {
+	printf("[%s] result %d\n", __FUNCTION__, (int)result);
+}
 
-int avs_test_main(int argc, char **argv)
+static int wifi_connect()
 {
-#ifdef CONFIG_EXAMPLES_AVS_TEST_CXXINITIALIZE
-	up_cxxinitialize();
-#endif	
-	
-	wifi_manager_result_e ret = WIFI_MANAGER_FAIL;
-	wifi_manager_ap_config_s config;
+	static wifi_manager_cb_s wifi_callbacks = {
+		wifi_sta_connected,
+		wifi_sta_disconnected,
+		NULL,
+		NULL,
+		NULL,
+	};
 
-	printf("wifi_manager_init()...\n");
-	ret = wifi_manager_init(&wifi_callbacks);
-	SAY_OK_OR_ERROR(WIFI_MANAGER_SUCCESS);
+	wifi_manager_result_e ret = wifi_manager_init(&wifi_callbacks);
+	if (ret != WIFI_MANAGER_SUCCESS) {
+		printf("wifi_manager_init failed, ret %d\n", (int)ret);
+		return -1;
+	}
 
+	wifi_manager_ap_config_s config;
 	config.ssid_length = strlen(CONFIG_EXAMPLES_AVS_TEST_SSID);
 	config.passphrase_length = strlen(CONFIG_EXAMPLES_AVS_TEST_PASSPHRASE);
 	strncpy(config.ssid, CONFIG_EXAMPLES_AVS_TEST_SSID, config.ssid_length + 1);
 	strncpy(config.passphrase, CONFIG_EXAMPLES_AVS_TEST_PASSPHRASE, config.passphrase_length + 1);
 	config.ap_auth_type = (wifi_manager_ap_auth_type_e)CONFIG_EXAMPLES_AVS_TEST_AUTHENTICATION;
 	config.ap_crypto_type = (wifi_manager_ap_crypto_type_e)CONFIG_EXAMPLES_AVS_TEST_CRYPTO;
 
-	printf("AP config: %s(%d) %s(%d) %d, %d\n",
-		config.ssid,
-		config.ssid_length,
-		config.passphrase,
-		config.passphrase_length,
-		config.ap_auth_type,
-		config.ap_crypto_type
-	);
-
-	/* current wifi mode is station, then this try will succeed */
-	printf("wifi_manager_connect_ap()...\n");
+	WIFI_CONNECTED_RESETFLAG();
+
+	printf("wifi_manager_connect_ap...\n");
 	ret = wifi_manager_connect_ap(&config);
-	SAY_OK_OR_ERROR(WIFI_MANAGER_SUCCESS);
+	if (ret != WIFI_MANAGER_SUCCESS) {
+		printf("wifi_manager_connect_ap failed, ret %d\n", (int)ret);
+		wifi_manager_deinit();
+		return -1;
+	}
+
+	printf("WIFI_CONNECTED_WAIT...\n");
+
+	WIFI_CONNECTED_WAIT();
+	if (!WIFI_CONNECTED_CHECKFLAG()) {
+		printf("wifi connect failed, timeout!\n");
+		wifi_manager_deinit();
+		return -1;
+	}
+
+	return 0;
+}
+
+static int wifi_disconnect()
+{
+	wifi_manager_result_e ret = wifi_manager_deinit();
+	if (ret != WIFI_MANAGER_SUCCESS) {
+		printf("wifi_manager_deinit failed, ret %d\n", (int)ret);
+		return -1;
+	}
+
+	return 0;
+}
+
+void *avs_test_entry(void *param)
+{
+#ifdef CONFIG_EXAMPLES_AVS_TEST_CXXINITIALIZE
+	up_cxxinitialize();
```

Comment:
if you set "CONFIG_HAVE_CXXINITIALIZE"
then you don't need it
because This is called automatically by os_bringup.

Reply:
Fixed, thanks.